.devcontainer/Dockerfile

```diff
@@ -3,17 +3,14 @@
 ARG VARIANT="ubuntu-22.04"
 FROM mcr.microsoft.com/vscode/devcontainers/base:0-${VARIANT}
 
-# Setup MS package signing key:
-# NB: needs changing if VARIANT changes
-RUN wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb && \
```

Comment:
Double-checking: the mismatch here between `20.04` and the `VARIANT` was _not_ the only problem, correct? I'm assuming that trying to update this is how you surfaced the problem.

Then a question: aren't we installing the Microsoft repo elsewhere, e.g. on VMs? Would it be better to try to make sure always install that on the devcontainer, use the same sources, &c? My main worry is that one of the packages might be split into more granular packages, and so they stop being 1-1 replacements.

Reply:
Second one:

I don't think we are using the MS packages anywhere else. I think we have two scenarios:

- For the Azure Functions service we rely on the preinstalled dotnet binaries (we build a in what is called a "framework-dependent" mode), so we already relinquish control over that.

- For the code that runs on the VMs (the new dotnet tasks), you've set them to build in self-contained mode, so the runtime is bundled with the binaries and we will not use any external dotnet runtime if it exists on the VMs.

Also worth noting is that the devcontainer does not affect our build at all (at the moment), it is purely for development purposes. What version of dotnet we are building against is inherited from the `ubuntu-18.04` image that our Github Actions is based upon. Getting this all aligned so we are using the exact same version everywhere would be worth doing but I'm not attempting that at the moment.

Reply:
We do have some external tools being added (like SharpFuzz) which will use the platform-installed dotnet tooling, but I don't think we have anything on the VMs which we build ourselves. I'll ask around :)